src/modules/settings/components/Table.jsx

```diff
@@ -171,22 +182,47 @@ function EditTable({options, setValue, value, ...props}) {
     return parseInt(Object.keys(data)[Object.keys(data).length - 1], 10) + 1 || 0;
   }
 
-  function addRow() {
+  function createRow() {
     const id = nextId();
-    const newRow = {id, status: Status.EDIT};
-    for (const header of options) {
-      switch (header.type) {
+    const row = {};
+
+    row.id = id;
+    row.status = Status.EDIT;
+
+    for (const option of options) {
+      switch (option.type) {
         case Types.STRING:
-          newRow[header.name] = '';
+          row[option.name] = '';
           break;
         case Types.DROPDOWN:
-          newRow[header.name] = header.defaultOption;
+          row[option.name] = option.defaultOption;
           break;
         default:
           break;
       }
     }
-    data[id] = newRow;
+
+    return row;
+  }
+
+  function addEmptyRow() {
+    const newRow = createRow();
+    data[newRow.id] = newRow;
+    setData({...data});
+  }
+
+  function handlePaste(event, id, dataKey) {
+    const paste = (event.clipboardData || window.clipboardData).getData('text');
+    const words = paste.split('\n');
```

Comment:
these are lines, not words.

we also want to likely trim and filter the results:

```
const lines = paste.split('\n').map(line => line.trim()).filter(line => line.length > 0);
```